Update examples and docs for backup locations

nrb opened the pull request:
Backup storage locations were not mentioned in our docs, and the ark server will fail to start if the default doesn't exist.

Signed-off-by: Nolan Brubaker <nolan@heptio.com>

nrb commented:
Actually, don't merge yet - still need to edit config-definition.md to not reference the fields that moved

skriss commented on examples/aws/00-ark-backupstoragelocation.yaml:
```diff
@@ -0,0 +1,32 @@
+    # e.g. "my-restic-bucket" or "my-restic-bucket/repos".
+    # This MUST be a different bucket than the main Ark bucket
+    # specified just above.
+    # restic-location: <YOUR_RESTIC_LOCATION>
```
this is now under `config`, but nothing else should've changed

nrb commented:
Fixed restic location for AWS, moved the backup storage location definitions out of the config-definition.md file into their own, and updated the IBM docs.

skriss commented on docs/aws-config.md:
```diff
@@ -139,9 +139,9 @@ kubectl create secret generic cloud-credentials \
 Specify the following values in the example files:
 
-* In `examples/aws/00-ark-config.yaml`:
+* In `examples/aws/00-ark-config.yaml` and `examples/aws/00-ark-backupstoragelocation.yaml`:
```
I think the idea with the number prefix on each YAML file was to have a unique one per file (00, 10, 20, 30, ...) although TBH I can't even remember why we did this - presumably to specify an order - @ncdc you remember the details?

nrb replied:
Order makes sense; would 01 be better for this than 00?

skriss replied:
yeah I think that'd be fine.  Or you could go `05` if you want to leave some room on either side in case something else comes up that needs to go in between (unlikely, but who knows)

skriss commented on docs/azure-config.md:
```diff
@@ -135,7 +135,11 @@ Now that you have your Azure credentials stored in a Secret, you need to replace
+
+* In file `examples/azure/00-ark-backupstoragelocation.yaml`:
+
+  * Replace `<YOUR_BUCKET>`.
```
This will need to be updated per #790 but I'll handle that.

skriss commented on docs/backupstoragelocation-definition.md:
```diff
@@ -0,0 +1,65 @@
+
+Ark can store backups in a number of locations. These are represented in the cluster via the `BackupStorageLocation` CRD.
+
+Ark needs a minimum of one `BackupStorageLocation`, named `default`, in order to store backups. Backups that do not explicitly specify a storage location will be saved to the `default` `BackupStorageLocation`.
```
Probably worth noting that by default, the default location is named "default", but you can override the default name with a server flag.  Good luck wording that :)

nrb replied:
Here's a few different wordings that I came up with:

1. Ark needs a minimum of one `BackupStorageLocation`. Without any changes, this will be expected to be named `default`, however  the expected name can be changed with the `--default-backup-storage-location` flag on the `ark server` command. Backups that do not explicitly specify a storage location will be saved to this `BackupStorageLocation`.


1. Ark must have a minimum of one `BackupStorageLocation`. By default, Ark expects this to be named `default`. The name can be changed by specifying `--default-backup-storage-location` on `ark server`. Backups that do not explicitly specify a storage location will be saved to this `BackupStorageLocation`.


1. Ark must have at least one `BackupStorageLocation`. By default, this is expected to be named `default`, however the name can be changed by specifying `--default-backup-storage-location` on `ark server`.  Backups that do not explicitly specify a storage location will be saved to this `BackupStorageLocation`.

skriss replied:
I like (3)

skriss commented on docs/backupstoragelocation-definition.md:
```diff
@@ -0,0 +1,65 @@
+
+#### Azure
+
+No parameters required.
```
I'll update this for #790

nrb replied:
Cool - yeah, I think either this gets handled in 790 or after both this and 790 get merged.

skriss commented on docs/gcp-config.md:
```diff
@@ -112,7 +112,7 @@ _Note: If you use a custom namespace, replace `heptio-ark` with the name of the
 Specify the following values in the example files:
 
-* In file `examples/gcp/00-ark-config.yaml`:
+* In file `examples/gcp/00-ark-backupstoragelocation.yaml`:
```
renumber

skriss commented on docs/ibm-config.md:
```diff
@@ -53,9 +53,9 @@ kubectl create secret generic cloud-credentials \
 Specify the following values in the example files:
 
-* In `examples/ibm/00-ark-config.yaml`:
+* In `examples/ibm/00-ark-backupstoragelocation.yaml`:
```
renumber

skriss commented on docs/aws-config.md:
```diff
@@ -139,9 +139,9 @@ kubectl create secret generic cloud-credentials \
 Specify the following values in the example files:
 
-* In `examples/aws/00-ark-config.yaml`:
+* In `examples/aws/00-ark-config.yaml` and `examples/aws/05-ark-backupstoragelocation.yaml`:
```
I like splitting this into 2 bullets like you did in `azure-config.md`, one for `Config` and one for `BackupStorageLocation`.  Then you can link to the Config definition doc for the config bullet, and the BSL definition doc for the BSL bullet.

skriss commented on docs/azure-config.md:
```diff
@@ -135,7 +135,11 @@ Now that you have your Azure credentials stored in a Secret, you need to replace
+
+* In file `examples/azure/05-ark-backupstoragelocation.yaml`:
+
+  * Replace `<YOUR_BUCKET>`.
```
I'd add a `See the BSL definition for details.` here

nrb commented:
@skriss Updated it and made the providers more consistent.

skriss commented on docs/aws-config.md:
```diff
@@ -139,9 +139,13 @@ kubectl create secret generic cloud-credentials \
 Specify the following values in the example files:
 
-* In `examples/aws/00-ark-config.yaml`:
+* In `examples/aws/00-ark-config.yaml` and `examples/aws/05-ark-backupstoragelocation.yaml`:
```
I think you want to drop "and `examples/aws/05-....` from this line, right?
